eras/conway/impl/src/Cardano/Ledger/Conway/PParams.hs

```diff
@@ -1159,10 +1165,15 @@ instance FromJSON (UpgradeConwayPParams Identity) where
         <*> o .: "dRepDeposit"
         <*> o .: "dRepActivity"
         <*> o .: "minFeeRefScriptCostPerByte"
+        <*> do
+          cm <- o .: "plutusV3CostModel"
+          case mkCostModel PlutusV3 cm of
+            Left e -> fail $ show e
+            Right cm' -> pure cm'
```

Comment:
You could rewrite this as 
`either (fail . show) pure (mkCostModel PlutusV3 cm)` if you fancied.